File: backend/app/schemas/export.py
"""
export.py — Pydantic v2 schemas for export requests.
"""
from enum import Enum

from pydantic import BaseModel


class ExportFormat(str, Enum):
    """Supported export formats."""
    md = "md"
    txt = "txt"
    pdf = "pdf"
    docx = "docx"
    zip = "zip"


class ExportRequest(BaseModel):
    """Schema for export request."""
    format: ExportFormat = ExportFormat.md

File: frontend/src/types/export.ts
/**
 * export.ts — Export format type definitions.
 * @module types/export
 */

export const ExportFormat = {
  Markdown: 'md',
  Text: 'txt',
  PDF: 'pdf',
  DOCX: 'docx',
  ZIP: 'zip',
} as const

export type ExportFormat = (typeof ExportFormat)[keyof typeof ExportFormat]
